### ISSUE TITLE
Basic Stacked Columns + linq

### ISSUE THREAD

**Issue by RiderOfTheli** [beto-rodriguez/Live-Charts#269]
Thank you for support in #261 , it works for me. 

I have one more question regarding that solution - how to implement it to Basic Stacked Columns? 
Should I split the result into e.g Dates and then add it to StackedColumnSeries?

I have a table/entity where I have 3 Columns -  LogDate, Activity, ActivityCount

I would like to present stacked columns per date with Activities. (ActivityCount is a double, so it will be a value for each activity)

Thank you in advance

**Comment by beto-rodriguez** [comment]
Hello

Can I please see what you have tried and why it is not working so I can hep you better?

Thanks!

**Comment by RiderOfTheli** [comment]
Hello,

I was trying to do this for StackBar and LineSeries:

this is my method:
```
`using (PcappConnection entityconn = new PcappConnection()) {
	Mapper = Mappers.Xy<MetricsTop5ActivitiesAVG>().X((prod, index) => index).Y(prod => prod.Elaps);
	dynamic records = entityconn.MetricsTop5ActivitiesAVG(EESelectedDG()).ToArray();
	ResultsAVG = records.AsChartValues();
	LabelsAVG = new ObservableCollection<string>(records.Select(x => x.Activity));

	AVGSeriesCollection = new SeriesCollection { new LineSeries {
		Title = "AVG",
		Values = new ChartValues<int> { records }
	} };
	YFormatter = value => value.ToString("N");
}
```

Well, I'm still playing with it, so it may look a little bit messy.
and xaml:
```
<lvc:CartesianChart Series="{Binding TeamLeaderViewM.AVGSeriesCollection}" LegendLocation="Right" >
<lvc:CartesianChart.AxisY>
<lvc:Axis Title="Average" LabelFormatter="{Binding TeamLeaderViewM.YFormatter}"></lvc:Axis>
</lvc:CartesianChart.AxisY>
<lvc:CartesianChart.AxisX>
<lvc:Axis Title="Day" Labels="{Binding TeamLeaderViewM.LabelsAVG}"></lvc:Axis>
</lvc:CartesianChart.AxisX>
 </lvc:CartesianChart>
```
When it comes to [http://lvcharts.net/App/examples/v1/wpf/Using%20LINQ](url) solution, it works for standard bars and it is okay with me. But I would like to extend this and use it for StackBars and LineSeries as well.

Thank you for your support

**Comment by beto-rodriguez** [comment]
Hello, and what is the error? which is your expected result?

**Comment by RiderOfTheli** [comment]
Hello,

Basically I would like to present these data stacked in 5 columns (X Axis will be a Day name, Y Axis count per activity). Like you have here in your example: [http://lvcharts.net/App/examples/v1/wpf/Basic%20Stacked](url). 

I was trying to do this via Mapper, like:
`MapperActStacked = Mappers.Xy<MetricsActivitiesPerDay>().X((x, index) => x.Activity).Y(mon => mon.CAct);`
but `**x.Activity**` in `.X((x, index) => x.Activity)` is a String and I cannot push it to Double. 
Thank you in advance for your help

**Comment by beto-rodriguez** [comment]
Can I see an example of MetricsActivitiesPerDay class, the definition and an example with its values would be useful, also let me know the properties you need to plot please

**Comment by RiderOfTheli** [comment]
MetricsActivitiesPerDay:

```
public partial class MetricsActivitiesPerDay
{
	public string Activity { get; set; }
	public Nullable<int> CAct { get; set; }
	public string Dayn { get; set; }
}
```
(I'm using Entity Framework, result taken from Select Statement):
![image](https://cloud.githubusercontent.com/assets/8041023/17566133/7a21cd38-5f3a-11e6-87c1-a1e2d67dd689.png)

CAct is a Count of Activity for particular day.
AxisX - Dayn as a Labels
AxisY - CAct per Activity in one bar

Hope this will help
Thank you!

**Comment by beto-rodriguez** [comment]
Ok, I tired what come to my mind:

    var mapper = Mappers.Xy<Vm>()
                .X(vm => classify(vm.Dayn))
                .Y(vm => vm.CAct ?? 0);

            Series = new SeriesCollection(mapper);

            Series.AddRange(source.GroupBy(x => x.Activity)
                .Select(x => new StackedColumnSeries
                {
                    Title = x.Key,
                    Values = x.AsChartValues()
                }).ToList());

it "works" but is bugged, I get some strange result, the data is classified correctly as far as I can see but the chart display seems bugged, I will investigate further.

**Comment by RiderOfTheli** [comment]
Thank you very much. Yes, I have the result here. Few of theme are 'flying' in the middle of the chart. I will play with it as much as I can.

Thank you!

**Comment by joselea** [comment]
I also have a similar issue working with StackedColumns, it sometimes shows an empty space and then columns floating on top.

![stackedcolumn isuue](https://cloud.githubusercontent.com/assets/20803827/17754096/cb9d4f46-6487-11e6-8b59-16ef4e9988a7.png)

**Comment by beto-rodriguez** [comment]
@RiderOfTheli @joselea It should be already working correctly, sadly I really want to get everything ready with high performance before doing a new release (need one week aprox I guess).

Does this works for you?

**Comment by joselea** [comment]
Do you think this fix can go out with the high performance release?

**Comment by beto-rodriguez** [comment]
it will be included in the next release, 0.7.12
